include/xgboost/c_api.h

```diff
@@ -198,23 +198,28 @@ XGB_DLL int XGDMatrixCreateFromCSR(char const *indptr, char const *indices, char
 XGB_DLL int XGDMatrixCreateFromDense(char const *data, char const *config, DMatrixHandle *out);
 
 /*!
- * \brief create a matrix content from CSC format
- * \param col_ptr pointer to col headers
- * \param indices findex
- * \param data fvalue
- * \param nindptr number of rows in the matrix + 1
- * \param nelem number of nonzero elements in the matrix
- * \param num_row number of rows; when it's set to 0, then guess from data
+ * \brief Create a matrix from a CSC matrix.
+ * \param indptr  JSON encoded __array_interface__ to column pointers in CSC.
+ * \param indices JSON encoded __array_interface__ to row indices in CSC.
+ * \param data    JSON encoded __array_interface__ to values in CSC.
+ * \param nrow     number of rows in the matrix.
+ * \param config  JSON encoded configuration.  Supported values are:
+ *   - missing: Which value to represent missing value.
+ *   - nthread (optional): Number of threads used for initializing DMatrix.
  * \param out created dmatrix
  * \return 0 when success, -1 when failure happens
  */
-XGB_DLL int XGDMatrixCreateFromCSCEx(const size_t* col_ptr,
-                                     const unsigned* indices,
-                                     const float* data,
-                                     size_t nindptr,
-                                     size_t nelem,
-                                     size_t num_row,
-                                     DMatrixHandle* out);
+XGB_DLL int XGDMatrixCreateFromCSC(char const *indptr, char const *indices, char const *data,
+                                   bst_ulong nrow, char const *c_json_config, DMatrixHandle *out);
+
+/*!
+ * \brief create a matrix content from CSC format
+ * \deprecated since 2.0.0
+ * \see XGDMatrixCreateFromCSC()
+ */
+XGB_DLL int XGDMatrixCreateFromCSCEx(const size_t *col_ptr, const unsigned *indices,
```

Comment:
Can we mark `XGDMatrixCreateFromCSREx` as deprecated too?

Reply:
done.